lib/diagnostic.js

```diff
@@ -48,8 +51,12 @@ const select = function (/* array, callback */) {
 // for which invoking the callback returns a truthy value
 // you may use any feature of JavaScript
 // other than `Array.prototype.find`
-const first = function (/* array, callback */) {
-
+const first = function (array, callback) {
+  for (let i = 0; i < array.length; i++) {
+    if (callback === array[i]) {
```

Comment:
You have a similar issue here as you did in question 3, but you're closer!
This should be:
```js
if(callback(array[i]){
  return array[i]
}
```
Callback is a function that needs to be run on each value of the array.